pkg/dashboard/dashboard.go

```diff
@@ -78,3 +115,84 @@ func Dashboard(opts Options) http.Handler {
 		writeTemplate(tmpl, opts, &data, w)
 	})
 }
+
+func calculateContainerCost(costPerCPUFloat float64, costPerGBFloat float64, c summary.ContainerSummary) float64 {
+	var cpuRequests, memRequests, cpuLimits, memLimits float64
+
+	cpuRequests = 0
+	memRequests = 0
+	cpuLimits = 0
+	memLimits = 0
+
+	if c.Limits != nil {
+		cpuLimits = float64(c.Limits.Cpu().Value())
+		memLimits = float64(c.Limits.Memory().Value())
+	}
+	if c.Requests != nil {
+		cpuRequests = float64(c.Requests.Cpu().Value())
+		memRequests = float64(c.Requests.Memory().Value())
+	}
+
+	cpuCost := costPerCPUFloat * getNonZeroAverage(cpuRequests, cpuLimits)
+	memCost := costPerGBFloat * ConvertToGB(int64(getNonZeroAverage(memRequests, memLimits)))
+
+	return toFixed(cpuCost+memCost, 4)
+}
+
+func getNonZeroAverage(req, limit float64) float64 {
+	if req == 0.0 {
+		return limit
+	}
+	if limit == 0.0 {
+		return req
+	}
+	return (req + limit) / 2.0
+}
+
+func calculateRecommendedCosts(costPerCPUFloat float64, costPerGBFloat float64, containerCost float64, c summary.ContainerSummary) (float64, float64) {
+	guaranteedCpuCostRecommended := costPerCPUFloat * float64(c.Target.Cpu().Value())
+	guaranteedMemCosttRecommended := costPerGBFloat * ConvertToGB(c.Target.Memory().Value())
+
+	burstableCpuCostRecommended := costPerCPUFloat * (float64(c.LowerBound.Cpu().Value() + c.UpperBound.Cpu().Value())) / 2
+	burstableMemCosttRecommended := costPerGBFloat * (ConvertToGB(c.LowerBound.Memory().Value()) + ConvertToGB(c.UpperBound.Memory().Value())) / 2
+
+	guaranteedCostRecommended := math.Abs(containerCost - (guaranteedCpuCostRecommended + guaranteedMemCosttRecommended))
+	burstableCostRecommended := math.Abs(containerCost - (burstableCpuCostRecommended + burstableMemCosttRecommended))
```

Comment:
This is not quite right - we don't want an absolute value.

If containerCost > recommendedCost, it should be negative (i.e. you'll save money by taking the recommendations)

Reply:
I updated the code. Thank you so much!